src/bindings/g_object/object.cr

```diff
@@ -7,6 +7,14 @@ module GObject
   class Object
     macro inherited
       {% unless @type.annotation(GObject::GeneratedWrapper) %}
+        macro method_added(method)
+          {% verbatim do %}
+            {% if method.name.starts_with?("do_") %}
```

Comment:
This is too generic and can cause problems for user code writing methods named `do_something` when there's no `something` function.

Would be nice to have a predefined list of virtual methods here, but to achieve this this macro need to go to `object.ecr`

another option is to have a `Virtual` annotation, but IMO the less annotations needed the better.